File: rules-interpreter/src/main/java/com/github/iamrenny/ruleflow/evaluators/BinaryOrContextEvaluator.java
package io.github.iamrenny.ruleflow.evaluators;

import io.github.iamrenny.ruleflow.RuleFlowLanguageParser;
import io.github.iamrenny.ruleflow.visitors.Visitor;

public class BinaryOrContextEvaluator implements ContextEvaluator<RuleFlowLanguageParser.BinaryOrContext> {

    @Override
    public Boolean evaluate(RuleFlowLanguageParser.BinaryOrContext ctx, Visitor visitor) {
        return (Boolean) visitor.visit(ctx.left) || (Boolean) visitor.visit(ctx.right);
    }
}

File: rules-interpreter/src/main/kotlin/com/rappi/analang/evaluators/BinaryOrContextEvaluator.kt
package com.github.iamrenny.rulesflow.evaluators

import com.rappi.fraud.analang.ANAParser
import com.github.iamrenny.rulesflow.visitors.Visitor

class BinaryOrContextEvaluator : ContextEvaluator<ANAParser.BinaryOrContext> {

    override fun evaluate(ctx: ANAParser.BinaryOrContext, visitor: Visitor): Boolean {
        return visitor.visit(ctx.left) as Boolean || visitor.visit(ctx.right) as Boolean
    }
}